py/client2/deephaven_client.pyx

```diff
@@ -0,0 +1,393 @@
+        nameBytes = name.encode()
+        result = deref(self.table).getColumn(nameBytes, strict)
+        if (result == NULL):
+            return None
+        return ColumnSource.create(move(result))
+
+    def getColumnIndex(self, name: unicode, strict: bool) -> size_t:
+        nameAsString = <string>name.encode()
+        return deref(self.table).getColumnIndex(nameAsString, strict)
+
+    def getRowSequence(self) -> RowSequence:
+        result = deref(self.table).getRowSequence()
+        return RowSequence.create(move(result))
+        
+    @property
+    def numRows(self) -> size_t:
+        return deref(self.table).numRows()
+
+    @property
+    def numColumns(self) -> size_t:
+        return deref(self.table).numColumns()
+
+    def toString(self, wantHeaders: bool, wantRowNumbers: bool, rowSequence = None) -> str:
+        if rowSequence is None:
+            result = deref(self.table).toString(wantHeaders, wantRowNumbers)
+        elif isinstance(rowSequence, list):
+            print("TODO")
+        elif isinstance(rowSequence, RowSequence):
+            result = deref(self.table).toString(wantHeaders, wantRowNumbers, (<RowSequence>rowSequence).rowSequence)
+        return result.decode()
+    
+# Simple wrapper of the corresponding C++ RowSequence class.
+cdef class RowSequence:
+    cdef shared_ptr[CRowSequence] rowSequence
+
+    @staticmethod
+    cdef RowSequence create(shared_ptr[CRowSequence] rowSequence):
+        result = RowSequence()
+        result.rowSequence = move(rowSequence)
+        return result;
+
+    def take(self, size: size_t) -> RowSequence:
+        rowSequence = deref(self.rowSequence).take(size)
+        return RowSequence.create(move(rowSequence))
+
+    def drop(self, size: size_t) -> RowSequence:
+        rowSequence = deref(self.rowSequence).drop(size)
+        return RowSequence.create(move(rowSequence))
+
+    @property
+    def size(self) -> size_t:
+        return deref(self.rowSequence).size()
+	
+    @property
+    def empty(self) -> bool:
+        return deref(self.rowSequence).empty()
+
+# A Cython fused type indicating all the valid element types for ColumnSource. For the C++
+# "string" type we use object. The C++ DateTime type is not yet implemented.
+ctypedef fused column_source_element_type_t:
+    bool
+    int8_t
+    int16_t
+    int32_t
+    int64_t
+    float
+    double
+    object
+
+# A Cython fused type indicating all the valid *primitive* element types for
+# ColumnSource, i.e. excluding object.
+ctypedef fused column_source_primitive_element_type_t:
+    bool
+    int8_t
+    int16_t
+    int32_t
+    int64_t
+    float
+    double
+
+# Confirm that the element type of destData is the same as the element type of Column Source.
+# Technically we don't need to do this check, as the C++ library has its own error checking,
+# but this gives us an opportunity to provide a more friendly error message here.
+cdef checkCompatibility(const CColumnSource &columnSource,
```

Comment:
I am not sure how visibility works here; AFAIK in python you typically prefix a function name with `_` when you want to make it "private" to a file.  Would that apply here?  Not sure if this funciton is intended for internal use only or is supposed to also be called by client code.